pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -386,6 +386,11 @@ func (r *shootReconciler) runReconcileShootFlow(ctx context.Context, o *operatio
 			Fn:           flow.TaskFn(botanist.DeployMetricsServer).RetryUntilTimeout(defaultInterval, defaultTimeout).SkipIf(o.Shoot.HibernationEnabled),
 			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, waitUntilOperatingSystemConfigReady, deployKubeScheduler),
 		})
+		_ = g.Add(flow.Task{
+			Name:         "Deploying VPNShoot system component",
+			Fn:           flow.TaskFn(botanist.DeployVPNShoot).RetryUntilTimeout(defaultInterval, defaultTimeout).SkipIf(o.Shoot.HibernationEnabled),
+			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, initializeShootClients, waitUntilOperatingSystemConfigReady, deployKubeScheduler),
```

Comment:
are these the correct dependencies? are there some more, or less?

Reply:
I think just `deployGardenerResourceManager` and `deployKubeScheduler` is enough since your newly introduced `vpnshoot` component will just create a `ManagedResource`, so there is no need to wait for `initializeShootClients` and `waitUntilOperatingSystemConfigReady`.